packages/macro/src/index.ts

```diff
@@ -110,54 +121,132 @@ function macro({ references, state, babel, config }: MacroParams) {
     }
   })
 
+  if (needsUseLinguiImport) {
+    addImport(babel, body, useLinguiImportModule, useLinguiImportName)
+  }
+
   if (needsI18nImport) {
-    addImport(babel, state, i18nImportModule, i18nImportName)
+    addImport(babel, body, i18nImportModule, i18nImportName)
   }
 
   if (jsxNodes.size) {
-    addImport(babel, state, TransImportModule, TransImportName)
+    addImport(babel, body, TransImportModule, TransImportName)
   }
 }
 
 function reportUnsupportedSyntax(path: NodePath, e: Error) {
   throw path.buildCodeFrameError(
-    `Unsupported macro usage. Please check the examples at https://lingui.dev/ref/macro#examples-of-js-macros. 
+    `Unsupported macro usage. Please check the examples at https://lingui.dev/ref/macro#examples-of-js-macros.
  If you think this is a bug, fill in an issue at https://github.com/lingui/js-lingui/issues
- 
+
  Error: ${e.message}`
   )
 }
 
+/**
+ * Pre-process useLingui macro
+ * 1. Get references to destructed t function for macro processing
+ * 2. Transform usage to non-macro useLingui
+ *
+ * @returns Array of paths to useLingui's t macro
+ */
+function processUseLingui(
+  path: NodePath,
+  newIdentifier: string
+): NodePath[] | null {
+  if (!path.parentPath.parentPath.isVariableDeclarator()) {
+    reportUnsupportedSyntax(
+      path,
+      new Error(
+        `\`useLingui\` macro must be used in variable declaration.
+        
+ Example:
+
+ const { t } = useLingui()
+`
+      )
+    )
+    return null
+  }
+
+  const varDec = path.parentPath.parentPath.node
```

Comment:
The code in this file is the "meat" of the macro transform.

I'd appreaciate if we could make this as readable as possible for future developers.



`varDec`: what does that mean? Can we give this a clearer name?

Reply:
its variable declarator, please ignore this in favor of https://github.com/lingui/js-lingui/pull/1867